samples/simpleasset/simpleasset.go

```diff
@@ -0,0 +1,238 @@
+package simpleasset
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+	"os"
+
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// SmartContract provides functions for managing an Asset
+type SmartContract struct {
+	contractapi.Contract
+}
+
+type Asset struct {
+	ID             string 			`json:"id"`
+	AssetType      string 			`json:"assettype"`
```

Comment:
I believe this field exists to help us identify an instance of this asset during an exchange? If so, I think the "type" characteristic ought to exist outside of the asset itself. Let this chaincode just manage the asset without worrying about what needs to be done for atomic swaps. When we instrument/adapt the code for HTLC, we can add type definitions, but outside the specs of the assets themselves.

Reply:
I and Krishna think asset type can be something independent of asset exchange right? What do you say?